### PR TITLE
refactor: Make some improvements to how often/when we announce a group

### PR DESCRIPTION
**Pull request by JFreegman**
Instead of announcing a group whenever our connection status changes, including when we gain or lose a TCP relay connection, we now only announce when our UDP status changes, or if our previously announced TCP relay has gone offline. We also refresh our announcement every 20 minutes regardless of any connection changes (this is to rotate the TCP relay in case the chosen one is somehow broken).

This change should vastly reduce the amount of unnecessary DHT traffic related to group announcements.

- - -
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/2348)

### CODE REVIEW

File: toxcore/group_chats.c [hunk 1]
```diff
@@ -7034,27 +7035,24 @@ static void do_gc_tcp(const GC_Session *c, GC_Chat *chat, void *userdata)
  * Updates our TCP and UDP connection status and flags a new announcement if our connection has
  * changed and we have either a UDP or TCP connection.
  */
-#define GC_SELF_CONNECTION_CHECK_INTERVAL 2
+#define GC_SELF_CONNECTION_CHECK_INTERVAL 5
```
Comment (sudden6) [inline toxcore/group_chats.c:7038]: Please add a comment in what unit these intervals are defined. I guess seconds, but I don't know.

File: toxcore/group_chats.c [hunk 2]
```diff
@@ -7034,27 +7035,24 @@ static void do_gc_tcp(const GC_Session *c, GC_Chat *chat, void *userdata)
...
-
-    const Messenger *m = c->messenger;
-    const unsigned int self_udp_status = ipport_self_copy(m->dht, &chat->self_ip_port);
-
+    const unsigned int self_udp_status = ipport_self_copy(c->messenger->dht, &chat->self_ip_port);
     const bool udp_change = (chat->self_udp_status != self_udp_status) && (self_udp_status != SELF_UDP_STATUS_NONE);
-    const bool tcp_change = tcp_connections != chat->tcp_connections;
 
-    if (is_public_chat(chat) && (udp_change || tcp_change)) {
+    if (is_public_chat(chat) &&
```
Comment (sudden6) [inline toxcore/group_chats.c:7050]: This condition is quite hard to parse, can you comment/simplify it?

**Review by sudden6 — COMMENTED** [review]
Generally looks good, just some documentation stuff

File: toxcore/group_chats.c (hunk 1 above)
Comment (JFreegman) [reply toxcore/group_chats.c]: Done.

File: toxcore/group_chats.c (hunk 2 above)
Comment (JFreegman) [reply toxcore/group_chats.c:7053]: Done.

**Review by sudden6 — APPROVED** [review]
